src/mscorlib/src/System/AppDomain.cs

```diff
@@ -837,10 +837,17 @@ public Object GetData(string name)
                 }
             }
         }
-        
+       
+#if !PLATFORM_UNIX        
```

Comment:
What was the actual problem with it on Unix? We have `GetCurrentThreadId` in the CoreCLR Unix PAL, so it should have worked just fine as is.